pkg/operator/hive/hiveadmission.go

```diff
@@ -142,8 +140,28 @@ func (r *ReconcileHiveConfig) deployHiveAdmission(hLog log.FieldLogger, h *resou
 			hLog.WithError(err).Error("error injecting certs")
 			return err
 		}
+	} else {
+		// Set the serving cert CA secret hash as an annotation on the pod template to force a rollout in the event it changes:
+		servingCertSecret := &corev1.Secret{}
+		if err := r.Client.Get(context.Background(), types.NamespacedName{Namespace: hiveNSName, Name: hiveAdmissionServingCertSecretName}, servingCertSecret); err != nil {
+			hLog.WithError(err).WithField("secretName", hiveAdmissionServingCertSecretName).Log(
+				controllerutils.LogLevel(err), "error getting serving cert secret")
+		}
+		hLog.Info("Running on OpenShift 4, hashing serving cert secret onto a hiveadmission deployment annotation")
+		certSecretHash := computeSecretDataHash(servingCertSecret.Data)
+		if hiveAdmDeployment.Spec.Template.Annotations == nil {
+			hiveAdmDeployment.Spec.Template.Annotations = map[string]string{}
+		}
+		hiveAdmDeployment.Spec.Template.Annotations[servingCertSecretHashAnnotation] = certSecretHash
```

Comment:
Does this need to be gated on whether Hive is running on OpenShift 4? Even when Hive is running on OpenShift 3.11 or another kube flavor, the same serving cert is in use.

Reply:
Good catch, I did not realize that.